spring-integration-core/src/main/java/org/springframework/integration/transformer/SimpleToAvroTransformer.java

```diff
@@ -41,6 +45,64 @@
 
 	private final EncoderFactory encoderFactory = new EncoderFactory();
 
+	private Expression typeIdExpression =
+			new FunctionExpression<Message<?>>((message) -> message.getPayload().getClass());
+
+	private EvaluationContext evaluationContext;
+
+	/**
+	 * Set the expression to evaluate against the message to determine the value
+	 * for the {@link AvroHeaders#TYPE} header.
+	 * @param expression the expression.
+	 * @return the transformer
+	 */
+	public SimpleToAvroTransformer typeExpression(Expression expression) {
+		assertExpressionNotNull(expression);
+		this.typeIdExpression = expression;
+		return this;
+	}
+
+	/**
+	 * Set the expression to evaluate against the message to determine the value
+	 * for the {@link AvroHeaders#TYPE} header.
+	 * @param expression the expression.
+	 * @return the transformer
+	 */
+	public SimpleToAvroTransformer typeExpression(String expression) {
+		assertExpressionNotNull(expression);
+		this.typeIdExpression = EXPRESSION_PARSER.parseExpression(expression);
+		return this;
+	}
+
+	/**
+	 * Set the expression to evaluate against the message to determine the value
+	 * for the {@link AvroHeaders#TYPE} header.
+	 * @param expression the expression.
+	 */
+	public void setTypeExpression(Expression expression) {
+		assertExpressionNotNull(expression);
+		this.typeIdExpression = expression;
+	}
+
+	/**
+	 * Set the expression to evaluate against the message to determine the value
+	 * for the {@link AvroHeaders#TYPE} header.
+	 * @param expression the expression.
+	 */
+	public void setTypeExpression(String expression) {
```

Comment:
This is not good for Java Beans introspector (used in Spring XML parsers). It just won't be able to determine which setter to use for incoming `String` value from the XML tag attribute.
That's why in other similar places such a setter is called with a `String` suffix.